### PR TITLE
changes in pytest markers: added order & dependency

### PR DESCRIPTION
**Pull request by bearecinos**
Well I did these changes to get rid of issue #22 but this issue magically went away when we merged @jrmaddison latest PR. 

Let me know if you are happy with anyways merging changes that I did to the pytest which are very minor but could prevent in the future issues when the test run in parallel. 

Changes are: 

- Ordering all the test under [test_runs.py](https://github.com/bearecinos/fenics_ice/blob/main/tests/test_runs.py) with order markers and dependencies. This require the installation of another pytest plugin (pytest-order) I can remove this dependency if you want as did not fix issue #22 but it might allow us in the future to play with the order of the test when being written.
- do a f.close() in read_vel_obs() ... as suggested by @jrmaddison before (it made no difference though everything only worked once I merged your PR) 

Let me know what you think ...

### CODE REVIEW

File: tests/test_runs.py [hunk 1]
```diff
@@ -314,12 +308,9 @@ def test_run_errorprop(existing_temp_model, monkeypatch, setup_deps, request):
                               work_dir,
                               'expected_Q_sigma_prior', tol=tol)
 
-@pytest.mark.dependency()
-@pytest.mark.runs
+@pytest.mark.dependency(["test_run_eigendec"],["test_run_errorprop"])
```
Comment (jrmaddison) [inline tests/test_runs.py:311]: Added an extra dependency here

File: tests/test_runs.py [hunk 2]
```diff
@@ -240,12 +238,10 @@ def forward_ts(cntrl, cntrl_init, name):
         assert(min_order > 1.95)
 
 
-@pytest.mark.dependency()
-@pytest.mark.runs
+@pytest.mark.order(3)
+@pytest.mark.dependency(['test_run_forward'], ['test_run_inversion'])
 def test_run_eigendec(existing_temp_model, monkeypatch, setup_deps, request):
```
Comment (jrmaddison) [inline tests/test_runs.py:243]: `request` not used?

File: tests/test_runs.py [hunk 3]
```diff
@@ -314,12 +308,9 @@ def test_run_errorprop(existing_temp_model, monkeypatch, setup_deps, request):
                               work_dir,
                               'expected_Q_sigma_prior', tol=tol)
 
-@pytest.mark.dependency()
-@pytest.mark.runs
+@pytest.mark.dependency(["test_run_eigendec"],["test_run_errorprop"])
 def test_run_invsigma(existing_temp_model, monkeypatch, setup_deps, request):
```
Comment (jrmaddison) [inline tests/test_runs.py:312]: `request` not used?

File: tests/test_runs.py [hunk 4]
```diff
@@ -131,12 +131,10 @@ def test_tv_run_inversion(persistent_temp_model, monkeypatch):
                                             seed=1.0e-5)
         assert(min_order > 1.95)
 
-@pytest.mark.dependency()
-@pytest.mark.runs
+@pytest.mark.order(2)
+@pytest.mark.dependency(["test_run_inversion"])
 def test_run_forward(existing_temp_model, monkeypatch, setup_deps, request):
```
[the remark this answers is not in the document]
Comment (bearecinos) [reply tests/test_runs.py:136]: yes that was already like that ... before I added the above lines should I take it away?

[the remark this answers is not in the document]
Comment (jrmaddison) [reply tests/test_runs.py:136]: Yes, I'd say it's better to remove unused variables

**Comment by bearecinos** [comment]
@jrmaddison deleted the request variable

**Comment by bearecinos** [comment]
@jrmaddison  Should we merge this then?

**Pull request merged**